cmd/awscollector/Dockerfile

```diff
@@ -1,19 +1,63 @@
+# The build mode (options: build, copy) passed in as a --build-arg. If build is specified, then the copy
+# stage will be skipped and vice versa. The build mode builds the binary from the source files, while
+# the copy mode copies in a pre-built binary.
+ARG BUILDMODE=build
```

Comment:
Should this be default as COPY for integration tests on CI? For docker-build and docker-build-arm should it be BUILDMODE=build (for first time user and it would be cache the layers after the first build)

Reply:
That could work too.

Reply:
Oops had missed this comment - think the default should be build so the Dockerfile can be built and run easily by external tools, e.g., `docker run https://github.com/aws-observability/aws-otel-collector`

copy is an interaction with our Makefile but may as well make the docker default as "docker-native" as possible

Reply:
Oh didn't know you could do that. Doesn't look like `docker run https://github.com/aws-observability/aws-otel-collector` is valid, but I see your point.